js/ui/grid_core/ui.grid_core.focus.js

```diff
@@ -423,9 +423,13 @@ module.exports = {
                 },
 
                 setFocusedRowIndex: function(rowIndex) {
+                    const dataController = this.getController('data');
+
                     this.callBase(rowIndex);
 
-                    const visibleRow = this.getController('data').getVisibleRows()[rowIndex];
+                    const visibleRowIndex = rowIndex - dataController?.getRowIndexOffset();
```

Comment:
Why is "?." operator here?
If dataController is null then visibleRowIndex will be NaN.